### PR TITLE
Lightning: Update Parameter Name from `tikv-importer.incremental-import` to `tikv-importer.parallel-import`

### LINKED ISSUE #45501: Update Parameter Name from 'tikv-importer.incremental-import' to 'tikv-importer.parallel-import'
## Background
The actual meaning of 'tikv-importer.incremental-import' is to use the physical import mode for parallel import, which requires the target table to be empty. However, the current name may mislead users into thinking it is for incremental import, leading to misuse.
When the user misuses it and sets the 'duplicate-resolution' to 'remove', conflicts between data in the source file and the existing data in the target table can lead to the mistaken deletion of the conflicting records in the target table.

## Enhancement
1) Starting from version 7.3, the parameter name has been changed from 'tikv-importer.incremental-import' to 'tikv-importer.parallel-import'.
2) If users pass the parameter name as 'tikv-importer.incremental-import', it will be converted to 'tikv-importer.parallel-import' for compatibility.

This improvement makes the functionality clearer and helps prevent user misuse.

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: close #45501 

Problem Summary:

The actual meaning of `tikv-importer.incremental-import` is to use the physical import mode for parallel import, which requires the target table to be empty. However, the current name may mislead users into thinking it is for incremental import, leading to misuse.
When the user misuses it and sets the `duplicate-resolution` to `remove`, conflicts between data in the source file and the existing data in the target table can lead to the mistaken deletion of the conflicting records in the target table.

### What is changed and how it works?

1. Starting from version 7.3, the parameter name has been changed from `tikv-importer.incremental-import` to `tikv-importer.parallel-import`.
2. If users pass the parameter name as `tikv-importer.incremental-import`, it will be converted to `tikv-importer.parallel-import` for compatibility.

Since there exist multiple unit tests and integration tests that have covered test cases of `tikv-importer.incremental-import`, I changed some of them to `tikv-importer.parallel-import`, and let others remain as `tikv-importer.incremental-import`. If all of the related unit tests and integration tests can pass, that means the functionality of both `tikv-importer.parallel-import` and `tikv-importer.incremental-import` work well.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Parameter name `tikv-importer.incremental-import` has been updated to `tikv-importer.parallel-import`. The meaning of this parameter is to use the physical import mode for parallel import, which requires the target table to be empty.
```

### CODE REVIEW

File: br/pkg/lightning/config/config.go [hunk 1]
```diff
@@ -1066,6 +1068,13 @@ func (t *TikvImporter) adjust() error {
 		return common.ErrInvalidConfig.GenWithStack("tikv-importer.backend must not be empty!")
 	}
 	t.Backend = strings.ToLower(t.Backend)
+	if !t.ParallelImport && t.IncrementalImport {
+		// need to assign t.IncrementalImport to t.ParallelImport when t.ParallelImport is false and t.IncrementalImport is true
+		t.ParallelImport = t.IncrementalImport
+	} else if t.ParallelImport && !t.IncrementalImport {
+		// need to assign t.ParallelImport to t.IncrementalImport when t.IncrementalImport is false and t.ParallelImport is true, to keep compatibility
+		t.IncrementalImport = t.ParallelImport
```
Comment (lance6716) [inline br/pkg/lightning/config/config.go:1076]: because IncrementalImport is deprecated, we don't need to update it anymore

Comment (lyzx2001) [reply br/pkg/lightning/config/config.go]: right, modified

**Review by lance6716 — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**